pkg/controller/masteruserrecord/sync.go

```diff
@@ -61,6 +87,55 @@ func (s *Synchronizer) synchronizeStatus() error {
 	return nil
 }
 
+// withClusterDetails returns the given user account status with additional information about
+// the target cluster such as API endpoint and Console URL if not set yet
+func (s *Synchronizer) withClusterDetails(status toolchainv1alpha1.UserAccountStatusEmbedded) (toolchainv1alpha1.UserAccountStatusEmbedded, error) {
+	if status.Cluster.Name != "" && (status.Cluster.APIEndpoint == "" || status.Cluster.ConsoleURL == "") {
+		status.Cluster.APIEndpoint = s.memberCluster.APIEndpoint
+		route := &routev1.Route{}
+		namespacedName := types.NamespacedName{Namespace: "openshift-console", Name: "console"}
+		err := s.memberCluster.Client.Get(context.TODO(), namespacedName, route)
+		if err != nil {
+			s.log.Error(err, "unable to get console route")
+			if kuberrors.IsNotFound(err) {
+				// It can happen if running in old OpenShift version like 3.x (minishift) in dev environment
+				// TODO do this only if run in dev environment
+				consoleURL, consoleErr := s.openShift3XConsoleURL(s.memberCluster.APIEndpoint)
+				if consoleErr != nil {
+					// Log the openShift3XConsoleURL() error but return the original missing route error
+					s.log.Error(err, "OpenShit 3.x web console unreachable", "url", consoleURL)
+					return status, errors.Wrapf(err, "unable to get web console route for cluster %s", s.memberCluster.Name)
+				}
+				s.log.Info("OpenShit 3.x web console URL used", "url", consoleURL)
+				status.Cluster.ConsoleURL = consoleURL
+				return status, nil
+			}
+			return status, err
+		}
+
+		status.Cluster.ConsoleURL = fmt.Sprintf("https://%s/%s", route.Spec.Host, route.Spec.Path)
+	}
+	return status, nil
+}
+
+// openShift3XConsoleURL checks if <apiEndpoint>/console URL is reachable.
+// This URL is used by web console in OpenShift 3.x
+func (s *Synchronizer) openShift3XConsoleURL(apiEndpoint string) (string, error) {
+	url := fmt.Sprintf("%s/console", apiEndpoint)
+	resp, err := consoleClient.Get(url)
+	if err != nil {
+		return url, err
+	}
+	if resp.StatusCode != http.StatusOK {
+		return url, errors.Errorf("status is not 200 OK: %s", resp.Status)
+	}
+	defer func() {
+		_, _ = ioutil.ReadAll(resp.Body)
+		_ = resp.Body.Close()
+	}()
```

Comment:
Should we put this right after the `Get` call? I think that we should read&close it also when status is not OK

Reply:
Oh.. good catch. We definitely should.

Reply:
Fixed in https://github.com/codeready-toolchain/host-operator/pull/103/commits/0bb6353313de16924fa58bc8666a5d191936c073